runtime/sema/type.go

```diff
@@ -859,6 +872,10 @@ func (t *GenericType) Resolve(typeArguments *TypeParameterTypeOrderedMap) Type {
 	return ty
 }
 
+func (t *GenericType) Map(_ common.MemoryGauge, f func(Type) Type) Type {
+	return f(t)
```

Comment:
Don't we need to create a new Generic type here?
Also, for completion, might want want to map `GenericType.TypeParameter.TypeBound`?